src/tools/concurrency/TracingActors.java

```diff
@@ -61,6 +60,29 @@ public final int getActorId() {
       throw new UnsupportedOperationException("Please remove this call and use getId instead");
     }
 
+    @Override
+    @TruffleBoundary
+    public synchronized void send(final EventualMessage msg,
+        final ForkJoinPool actorPool) {
+      super.send(msg, actorPool);
+      if (VmSettings.SENDER_SIDE_TRACING) {
+        msg.getTracingNode().record(this.version);
+        this.version++;
+        // TODO maybe try to get the recording itself done outside the synchronized method
```

Comment:
Is this TODO still current?

Reply:
I don't think this would change much, in a single threaded execution it doesn't matter. In a multithreaded execution it might reduce contention of an actor a bit, e.g. for philosophers, but i don't think the performance would change much.